packages/protocol/test-sol/devchain/e2e/common/Governance.t.sol

```diff
@@ -0,0 +1,431 @@
+  event LogConstitutionCase(ConstitutionCase);
+
+  // snapshot
+  uint256 internal constitutionSnapshot;
+
+  // parametrization
+  modifier parametrized__constitutionCase() {
+    for (uint256 i = 0; i < constitutionCases.length; i++) {
+      currentCase = constitutionCases[i];
+      if (constitutionSnapshot == 0) constitutionSnapshot = vm.snapshot();
+      _;
+      vm.revertTo(constitutionSnapshot);
+    }
+  }
+
+  function setUp() public virtual override {
+    // get contracts from constitution
+    string memory constitutionJson_ = vm.readFile("./governanceConstitution.json");
+    string[] memory contractNames_ = vm.parseJsonKeys(constitutionJson_, "");
+
+    // vars for looping
+    string memory contractName_;
+    address address_;
+    string memory functionName_;
+    bytes4 selector_;
+    uint256 threshold_;
+    string memory selectorJson_;
+    string[] memory functionsWithTypes_;
+    string[] memory functionNames_;
+
+    // loop over contract names
+    for (uint256 i = 0; i < contractNames_.length; i++) {
+      contractName_ = contractNames_[i];
+      if (contractName_.equals("proxy")) {
+        // skip proxy address
+        continue;
+      } else {
+        // set address from registry
+        address_ = registryContract.getAddressForStringOrDie(contractName_);
+      }
+
+      // load selectors for given contract from file
+      selectorJson_ = vm.readFile(string.concat("./.tmp/selectors/", contractName_, ".json"));
+
+      // get function names with types
+      functionsWithTypes_ = vm.parseJsonKeys(selectorJson_, "");
+
+      // get functions names from constitution for contract
+      functionNames_ = vm.parseJsonKeys(constitutionJson_, string.concat(".", contractName_));
+
+      // loop over function names
+      for (uint256 j = 0; j < functionNames_.length; j++) {
+        functionName_ = functionNames_[j];
+        if (functionName_.equals("default")) {
+          // use empty selector as default
+          selector_ = hex"00000000";
+        } else {
+          // retrieve selector from selector JSON
+          selector_ = selectorJson_.getSelector(functionsWithTypes_, functionName_, vm);
+        }
+
+        // determine treshold from constitution
+        threshold_ = constitutionJson_.readUint(
+          string.concat(".", contractName_, ".", functionName_)
+        );
+
+        // push new test case
+        constitutionCases.push(
+          ConstitutionCase(contractName_, address_, functionName_, selector_, threshold_)
+        );
+      }
+    }
+  }
+
+  function test_shouldHaveCorrectThreshold() public parametrized__constitutionCase {
+    emit LogConstitutionCase(currentCase);
+    assertEq(
+      governance.getConstitution(currentCase.contractAddress, currentCase.selector),
+      currentCase.threshold
+    );
+  }
+}
+
+contract E2E_Governance is Devchain {
+  using stdJson for string;
+
+  // config
+  address internal ownerAddress = 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266;
```

Comment:
any reason why not to use actor("owner") ?

Reply:
In `MigrationsConstants` there's defined `DEPLOYER_ACCOUNT = 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266`.

Which is used in `Migration.s.sol` as original owner for the contracts.
E2E tests inherit initial state from `Migration.s.sol` so I wanted to use above address.